core/arch/arm/tee/entry_fast.c

```diff
@@ -30,8 +28,8 @@ static void tee_entry_get_shm_config(struct thread_smc_args *args)
 
 static void tee_entry_fastcall_l2cc_mutex(struct thread_smc_args *args)
 {
-	TEE_Result ret;
 #ifdef ARM32
+	TEE_Result ret;
```

Comment:
The variable `ret` shall be initialized (c.f. [optee coding standards](https://optee.readthedocs.io/en/latest/general/coding_standards.html#coding-standards)). So what about:

```c
static void tee_entry_fastcall_l2cc_mutex(struct thread_smc_args *args)
{
	TEE_Result ret = TEE_ERROR_NOT_SUPPORTED;
#ifdef ARM32
	paddr_t pa = 0;

	switch (args->a1) {
	case OPTEE_SMC_L2CC_MUTEX_GET_ADDR:
		ret = tee_get_l2cc_mutex(&pa);
		reg_pair_from_64(pa, &args->a2, &args->a3);
		break;
	case OPTEE_SMC_L2CC_MUTEX_SET_ADDR:
		pa = reg_pair_to_64(args->a2, args->a3);
		ret = tee_set_l2cc_mutex(&pa);
		break;
	case OPTEE_SMC_L2CC_MUTEX_ENABLE:
		ret = tee_enable_l2cc_mutex();
		break;
	case OPTEE_SMC_L2CC_MUTEX_DISABLE:
		ret = tee_disable_l2cc_mutex();
		break;
	default:
		args->a0 = OPTEE_SMC_RETURN_EBADCMD;
		return;
	}
#endif
	if (ret == TEE_ERROR_NOT_SUPPORTED)
		args->a0 = OPTEE_SMC_RETURN_UNKNOWN_FUNCTION;
	else if (ret)
		args->a0 = OPTEE_SMC_RETURN_EBADADDR;
	else
		args->a0 = OPTEE_SMC_RETURN_OK;
}
```

Reply:
Coverity will still whine about the piece of code being dead. If ARM32=n, we got
```
        TEE_Result ret = TEE_ERROR_NOT_SUPPORTED;

	if (ret == TEE_ERROR_NOT_SUPPORTED)
		args->a0 = OPTEE_SMC_RETURN_UNKNOWN_FUNCTION;
	else if (ret) //DEAD
		args->a0 = OPTEE_SMC_RETURN_EBADADDR; //DEAD
	else //DEAD
		args->a0 = OPTEE_SMC_RETURN_OK; //DEAD
```

Reply:
Yet another situation where Coverity has a different objective.
We don't mind "dead" code, as long as it's used in at least one configuration. This is in fact something we encourage since it gives better compile test coverage.

Reply:
Well, I do understand Coverity scanning is not in your process and might conflict with some of your coding standards. However, on this particular case, we made sure `paddr_t pa` is within `#ifdef ARM32 ... #endif` why not doing the same thing with `TEE_Result ret` ?

Reply:
Ok